### PR TITLE
Fix HIDPI on pyglet

### LINKED ISSUE #2325: Apple M1 macOS 12 canvas is a quarter of window size
I have found an issue on macos 12 with apple m1 - dont know if it exists on x64 macos - dont have one so cannot test it.

all test are done with the offical examples!

The issue seems to be that the canvas is not the specified size - it's a quarter of it 
This is with pyglet
Untitled

pyglet scaled the whole canvas to to a quarter and also renders it at a quarter. Moving the qube works fine as you would expect just scaled down to a quarter

In contrast wx:

https://user-images.githubusercontent.com/13086508/164896545-086ae183-f87e-4934-b781-e3c296984bdd.mov

With wx the canvas is rendered a the correct resolution but only the lower left is quarter is visable. Moving the scene quarters the output to a quarter but the qube is now centered in frame - letting go of the mouse button goes back to the full canvas size with just the lower left quarter visable

The behavior continues with all of the other examples I tested

### PR DESCRIPTION
**Pull request by almarklein**
Closes #2325 

* [x] Fix for Pyglet (confirmed).
* [x] Fix for wx (confirmed).

### CODE REVIEW

**Comment by almarklein** [comment]
A note: the extra logic in rounding in wx was needed in pygfx, but this might not be the case for vispy.

File: vispy/app/backends/_wx.py [hunk 1]
```diff
@@ -376,6 +376,11 @@ def _vispy_get_size(self):
         w, h = self.GetClientSize()
         return w, h
 
+    def _vispy_get_physical_size(self):
+        lsize = self.GetClientSize()
+        ratio = self.GetContentScaleFactor()
+        return round(lsize[0] * ratio + 0.01), round(lsize[1] * ratio + 0.01)
```
Comment (djhoese) [inline vispy/app/backends/_wx.py:382]: I'm not 100% sure, but this may need to be integers...or at least they are supposed to be I think.

**Comment by almarklein** [comment]
I did not check other GUI backends, but it could be that more are missing the implementation of `_vispy_get_physical_size`. The solution would be similar though.

**Comment by brisvag** [comment]
@almarklein Did you plan to check the other backends here, or are you happy with this?

**Comment by almarklein** [comment]
Oh let's just merge this.

**Pull request merged**